### ISSUE TITLE
When you die as a Special Infected and respawn as a Normal Infected, you lose the maximum health buff but will overheal equal to that amount

### ISSUE THREAD

**Issue by BimboBumboBambo** [DFS-Servers/BugReportsAndSuggestions#126]
**Steam Profile**
https://steamcommunity.com/id/BimboBumboBambo/

**Describe the bug**
Whenever you go spy with the Conniver's Kunai equipped, you obviously are getting a health debuff. Nothing wrong with that. However, when you die as special infected(for example "Smoker") and you respawn back to a normal zombie, the health that you had as special infected turns into a temporary overheal. This ends up with spies who have over 200 health for a short amount of time.

**To Reproduce**

1. Go Spy with the Conniver's Kunai equipped. (@shinsasch - this happens on all knives)
2. Respawn as special infected(only Stalker and Smoker.)
3. Killbind or die normally.
4. Respawn as normal infected. You should now have a overheal buff.

**Expected behavior**
I expect this to be fixed in the next patch of SZF, so that matches can be more balanced again.

**Screenshots or error logs**
![glitch 1](https://user-images.githubusercontent.com/41446589/57570412-f4a8c000-7401-11e9-9a82-05334c74b447.jpg)
![glitch 2](https://user-images.githubusercontent.com/41446589/57570413-f6728380-7401-11e9-935e-7c3c4bbe932d.jpg)

**Server Info**
 - Region: EU
 - IP [e.g. 89.35.29.11:27145 ...]

**Additional context**
As I said, it's a temporary overheal buff. It should go away in seconds. Still, this gives spies a huge advantage. Especially those who know how to use him in szf.

**Comment by shinsasch** [comment]
Hi, I am aware of this issue myself, it is spot on except that it happens on ALL knives and ONLY on Spy. As of yet I am to find out why exactly the Spy is the sole class affected by this since I bound to health increase attribute to the client entity and not a weapon.

**Comment by shinsasch** [comment]
Nontheless, thanks for reporting and after it is addressed you will receive a cool box.

**Comment by BimboBumboBambo** [comment]
You're welcome. Thanks for the gamemode btw.

**Comment by BimboBumboBambo** [comment]
Also, one one of the maps "sfz_expedition_C2," zombies are able to stop the survivors from capture. I don't know if this is intentional or if it's an oversight.

**Comment by shinsasch** [comment]
Not an oversight, this is a map maker's choice because there are less Infected at the start to the moment the point to be captured (that's why camping with Sentry is so strong there, its so open, etc.)

I suggest that all players actually strive to stick together, to the control point asap and play defensively instead of dieing one by one, it will solve a lot of problems that are based on the Survivor team's lack of teamwork.

**Comment by BimboBumboBambo** [comment]
Alright. Thanks for letting me know.

**Comment by BimboBumboBambo** [comment]
I've just recently found this out.

Apparently, the Spy isn't the only one able to benefit from a temporary overheal buff. The Heavy does too, and this is how:

1. Equip the Gloves of Running Urgently
2. Spawn as special infected(for example Boomer)
3. Die
4. Gain a temporary health buff.

So far I only managed to see it happen with the GRU, so the health was drained back extremely quickly. I'll be letting you know if I can find anything else.
